tests/testthat/test-deparse.R

```diff
@@ -0,0 +1,78 @@
+context("deparse()")
+
+library(tibble)
+library(dplyr)
+
+test_that("deparse handles lists appropriately", {
+  check_deparse_identical(list(LETTERS))
+  check_deparse_identical(list(x = 1:4))
+  check_deparse_identical(list(list(list(1:4), y = 1:5)))
+
+  my_list <- list(x = 1:4)
+  attr(my_list, "my_attr") <- "test"
+
+  check_deparse_identical(my_list)
+
+  my_classed_list <- my_list
+  class(my_classed_list) <- "test_class"
+
+  check_deparse_identical(my_classed_list)
+})
+
+test_that("deparse handles basic types", {
+  check_deparse_identical(1:5)
+  check_deparse_identical(LETTERS)
+  check_deparse_identical(c(TRUE, FALSE))
+  check_deparse_identical(NULL)
+})
+
+test_that("deparse handles dates", {
+  check_deparse_identical(Sys.Date())
+  check_deparse_identical(as.POSIXct("2003-04-05 06:07:08 UTC"))
+  check_deparse_identical(as.POSIXlt("2003-04-05 06:07:08 UTC"))
+})
+
+test_that("deparse handles functions", {
+  check_deparse_equal(function(x) x + 1)
+  check_deparse_equal(mean)
+  expect_output(print(deparsec(function(x) x + 1)), "function \\(x\\) x \\+ 1")
```

Comment:
It breaks on some tests e.g.

    check_deparse_equal(tibble(x = 1:3, y = list(4:6, 7:9, 10:15)))
    # Error in equal_data_frame(target, current, ignore_col_order = ignore_col_order,  : 
    #  Can't join on 'y' x 'y' because of incompatible types (list / list)